core/src/main/scala/org/apache/spark/shuffle/IndexShuffleBlockResolver.scala

```diff
@@ -173,6 +183,13 @@ private[spark] class IndexShuffleBlockResolver(
    */
   override def putShuffleBlockAsStream(blockId: BlockId, serializerManager: SerializerManager):
       StreamCallbackWithID = {
+    // Throw an exception if we have exceeded maximum shuffle files stored
+    remoteShuffleMaxDisk.foreach { maxBytes =>
+      val bytesUsed = getShuffleBytesStored()
```

Comment:
This looks like a performance degradation a little due to the additional disk IO. Do we need this at every `putShuffleBlockAsStream`? Do you know how much does this affected, @holdenk ?

cc @dbtsai

Reply:
That's a good question. So looking at benchmarks folks have done for file size takes a few microseconds per file (and speeds up on subsequent runs due to cache). It doesn't need to actually read the file to determine the size (if it did that would be much slower).